src/openzeppelin/tests/access/ownable/test_dual_ownable.cairo

```diff
@@ -0,0 +1,203 @@
+use openzeppelin::tests::mocks::dual_ownable_mocks::CamelOwnableMock;
+use openzeppelin::tests::mocks::dual_ownable_mocks::SnakeOwnablePanicMock;
+use openzeppelin::tests::mocks::dual_ownable_mocks::CamelOwnablePanicMock;
+use openzeppelin::tests::mocks::non_implementing_mock::NonImplementingMock;
+
+// Other
+use openzeppelin::tests::utils;
+use openzeppelin::tests::utils::PanicTrait;
+
+///
+/// Constants
+///
+
+fn OWNER() -> ContractAddress {
+    contract_address_const::<10>()
+}
+
+fn NEW_OWNER() -> ContractAddress {
+    contract_address_const::<20>()
+}
+
+///
+/// Setup
+///
+
+fn setup_snake() -> (DualCaseOwnable, IOwnableDispatcher) {
+    let mut calldata = ArrayTrait::new();
+    set_caller_address(OWNER());
+    let target = utils::deploy(SnakeOwnableMock::TEST_CLASS_HASH, calldata);
+    (DualCaseOwnable { contract_address: target }, IOwnableDispatcher { contract_address: target })
+}
+
+fn setup_camel() -> (DualCaseOwnable, IOwnableDispatcher) {
+    let mut calldata = ArrayTrait::new();
+    set_caller_address(OWNER());
+    let target = utils::deploy(CamelOwnableMock::TEST_CLASS_HASH, calldata);
+    (DualCaseOwnable { contract_address: target }, IOwnableDispatcher { contract_address: target })
+}
+
+fn setup_non_ownable() -> DualCaseOwnable {
+    let calldata = ArrayTrait::new();
+    set_caller_address(OWNER());
+    let target = utils::deploy(NonImplementingMock::TEST_CLASS_HASH, calldata);
+    DualCaseOwnable { contract_address: target }
+}
+
+fn setup_ownable_panic() -> (DualCaseOwnable, DualCaseOwnable) {
+    set_caller_address(OWNER());
+    let snake_target = utils::deploy(SnakeOwnablePanicMock::TEST_CLASS_HASH, ArrayTrait::new());
+    let camel_target = utils::deploy(CamelOwnablePanicMock::TEST_CLASS_HASH, ArrayTrait::new());
+    (
+        DualCaseOwnable {
+            contract_address: snake_target
+            }, DualCaseOwnable {
+            contract_address: camel_target
+        }
+    )
+}
+
+///
+/// Case agnostic methods
+///
+
+#[test]
+#[available_gas(2000000)]
+fn test_dual_owner() {
+    let (snake_dispatcher, _) = setup_snake();
+    let (camel_dispatcher, _) = setup_camel();
+    assert(snake_dispatcher.owner() == OWNER(), 'Should return OWNER');
+    assert(camel_dispatcher.owner() == OWNER(), 'Should return OWNER');
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('ENTRYPOINT_NOT_FOUND', ))]
+fn test_dual_no_owner() {
+    let dispatcher = setup_non_ownable();
+    dispatcher.owner();
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('Some error', 'ENTRYPOINT_FAILED', ))]
+fn test_dual_owner_exists_and_panics() {
+    let (dispatcher, _) = setup_ownable_panic();
+    dispatcher.owner();
+}
+
+///
+/// snake_case target
+///
+
+#[test]
+#[available_gas(2000000)]
+fn test_dual_transfer_ownership() {
+    let (dispatcher, target) = setup_snake();
+    set_contract_address(OWNER()); // Bug with test-runner
```

Comment:
what bug exactly? the problem in ERC721 is only with the camelCase methods since it's after the second `call_contract_syscall` attempt only

Reply:
Correct, but we are still using the workaround `set_contract_address` for the first call as opposed to `set_caller_address`. This test in particular is setting the ownable contract address to the owner's address

Reply:
why do you say that's a bug? it makes sense to me that `call_contract_syscall` assumes to be in a contract's context when it's called and it forwards the current contract address as the caller.

Reply:
As discussed offline, setting the contract address as the caller in this context is _not_ a bug. Comments are removed and there's a tmp reference to the issue raised in cairo for the ignored `test_dual_transferOwnership` and `test_dual_renounceOwnership`